src/TimeSteppers/time_stepping_kernels.jl

```diff
@@ -93,24 +83,33 @@ end
 """ Apply boundary conditions by adding flux divergences to the right-hand-side. """
 function calculate_boundary_tendency_contributions!(Gⁿ, arch, U, C, clock, state)
 
+    events = []
+
     # Velocity fields
     for i in 1:3
-        apply_x_bcs!(Gⁿ[i], U[i], arch, clock, state)
-        apply_y_bcs!(Gⁿ[i], U[i], arch, clock, state)
-        apply_z_bcs!(Gⁿ[i], U[i], arch, clock, state)
+        x_bcs_event = apply_x_bcs!(Gⁿ[i], U[i], arch, clock, state)
+        y_bcs_event = apply_y_bcs!(Gⁿ[i], U[i], arch, clock, state)
+        z_bcs_event = apply_z_bcs!(Gⁿ[i], U[i], arch, clock, state)
+
+        push!(events, x_bcs_event, y_bcs_event, z_bcs_event)
     end
 
     # Tracer fields
     for i in 4:length(Gⁿ)
-        apply_x_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
-        apply_y_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
-        apply_z_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
+        x_bcs_event = apply_x_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
+        y_bcs_event = apply_y_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
+        z_bcs_event = apply_z_bcs!(Gⁿ[i], C[i-3], arch, clock, state)
+
+        push!(events, x_bcs_event, y_bcs_event, z_bcs_event)
     end
 
+    events = filter(e -> typeof(e) <: Base.Event, events)
```

Comment:
Why would `Base.Event` show up here?

Reply:
Does `KernelAbstractions` have it's own `Event` type? I didn't realize.